compiler/parser/python.lalrpop

```diff
@@ -1807,18 +1721,17 @@ ArgumentList: ArgumentList = {
     }
 };
 
-FunctionArgument: (Option<(ast::Location, ast::Location, Option<String>)>, ast::Expr) = {
+FunctionArgument: (Option<(crate::Location, crate::Location, Option<String>)>, ast::Expr) = {
     <location:@L> <e:NamedExpressionTest> <c:CompFor?> <end_location:@R> => {
         let expr = match c {
-            Some(c) => ast::Expr {
+            Some(c) => ast::Expr::new(
                 location,
-                end_location: Some(end_location),
-                custom: (),
-                node: ast::ExprKind::GeneratorExp {
+                end_location,
+                ast::ExprKind::GeneratorExp {
                     elt: Box::new(e),
                     generators: c,
                 }
-            },
+            ),
```

Comment:
I feel like all of the changes here could be upstreamed, or many of them, maybe not the `crate::Location`.

Reply:
It's plausible that we could upstream this entire change, but I'm not as familiar with the implications outside of the parser. I could tag in some folks on the RustPython side if we want though. How are you thinking about that?

Reply:
That could make our life significantly easier when pulling in upstream changes. So the idea is to upstream the reactors to use `Node::new` rather than `Node { ...}`

Reply:
See https://github.com/RustPython/RustPython/pull/4928